publish command fails when script path has spaces

System information:
```sh
λ dotnet --version
2.1.602
λ dotnet script --version
0.29.1
λ  systeminfo | findstr /B /C:"OS Name" /C:"OS Version"
OS Name:                   Microsoft Windows 10 Enterprise
OS Version:                10.0.16299 N/A Build 16299
```

Replicating:
```sh
λ dotnet script new test.csx
Creating 'test.csx'
...'C:\Users\ashley\test script\test.csx' [Created]

λ dotnet script test.csx
Hello world!

λ dotnet script publish test.csx
dotnet publish failed with result '1'
```

I was able to successfully publish the script when i moved it to a directory with no spaces.

This works:
```sh
λ mkdir ~/testscript && cd ~/testscript
λ dotnet script new test.csx
λ dotnet script publish test.csx
```

This does not work:
```sh
λ mkdir ~/test\ script && cd ~/test\ script
λ dotnet script new test.csx
λ dotnet script publish test.csx
```

Debug output:
```
λ dotnet script --debug publish test.csx
info: Dotnet.Script.Core.ScriptCompiler[0]
      Current runtime is 'win'.
dbug: Dotnet.Script.DependencyModel.ProjectSystem.ScriptProjectProvider[0]
      Creating project file for C:\Users\ashley\Documents\test script\test.csx
dbug: Dotnet.Script.DependencyModel.ProjectSystem.ScriptParser[0]
      Parsing C:\Users\ashley\Documents\test script\test.csx
dbug: Dotnet.Script.DependencyModel.ProjectSystem.ScriptProjectProvider[0]
      Project file saved to C:\Users\ashley\AppData\Local\Temp\scripts\C\Users\ashley\Documents\test script\script.csproj
dbug: Dotnet.Script.DependencyModel.ProjectSystem.ScriptProjectProvider[0]
      <?xml version="1.0" encoding="utf-8"?>
      <Project Sdk="Microsoft.NET.Sdk">
        <PropertyGroup>
          <OutputType>Exe</OutputType>
          <TargetFramework>netcoreapp2.1</TargetFramework>
          <LangVersion>latest</LangVersion>
        </PropertyGroup>
        <ItemGroup></ItemGroup>
      </Project>
dbug: Dotnet.Script.DependencyModel.ProjectSystem.ScriptProjectProvider[0]
      Generating NuGet config evaluated at C:\Users\ashley\Documents\test script to C:\Users\ashley\AppData\Local\Temp\scripts\C\Users\ashley\Documents\test script\NuGet.Config
dbug: Dotnet.Script.DependencyModel.Context.CachedRestorer[0]
      Found cached csproj file at: C:\Users\ashley\AppData\Local\Temp\scripts\C\Users\ashley\Documents\test script\script.csproj.cache
dbug: Dotnet.Script.DependencyModel.Context.CachedRestorer[0]
      Skipping restore. C:\Users\ashley\AppData\Local\Temp\scripts\C\Users\ashley\Documents\test script\script.csproj and C:\Users\ashley\AppData\Local\Temp\scripts\C\Users\ashley\Documents\test script\script.csproj.cache are identical.
dbug: Dotnet.Script.DependencyModel.Context.ProfiledRestorer[0]
      Restoring C:\Users\ashley\AppData\Local\Temp\scripts\C\Users\ashley\Documents\test script\script.csproj took 2ms
dbug: Dotnet.Script.Core.ScriptCompiler[0]
      Configuration/Optimization mode: Debug
dbug: Dotnet.Script.DependencyModel.Process.CommandRunner[0]
      Executing 'dotnet publish "C:\Users\ashley\AppData\Local\Temp\scripts\C\Users\ashley\Documents\test script\script.csproj" -c Release -r win10-x64 -o C:\Users\ashley\Documents\test script\publish\win10-x64'
dbug: Dotnet.Script.DependencyModel.Process.CommandRunner[0]
      Microsoft (R) Build Engine version 16.0.450+ga8dc7f1d34 for .NET Core
dbug: Dotnet.Script.DependencyModel.Process.CommandRunner[0]
      Copyright (C) Microsoft Corporation. All rights reserved.
dbug: Dotnet.Script.DependencyModel.Process.CommandRunner[0]
      MSBUILD : error MSB1008: Only one project can be specified.
dbug: Dotnet.Script.DependencyModel.Process.CommandRunner[0]
      Switch: script\publish\win10-x64
dbug: Dotnet.Script.DependencyModel.Process.CommandRunner[0]
      For switch syntax, type "MSBuild -help"
System.Exception: dotnet publish failed with result '1'
   at Dotnet.Script.Core.ScriptPublisher.CreateExecutable[TReturn,THost](ScriptContext context, LogFactory logFactory, String runtimeIdentifier) in C:\Users\VssAdministrator\AppData\Local\Temp\tmp794B\Dotnet.Script.Core\ScriptPublisher.cs:line 91
   at Dotnet.Script.Core.Commands.PublishCommand.Execute(PublishCommandOptions options) in C:\Users\VssAdministrator\AppData\Local\Temp\tmp794B\Dotnet.Script.Core\Commands\PublishCommand.cs:line 47
   at Dotnet.Script.Program.<>c__DisplayClass4_4.<Wain>b__14() in C:\Users\VssAdministrator\AppData\Local\Temp\tmp794B\Dotnet.Script\Program.cs:line 180
   at Dotnet.Script.Program.Wain(String[] args) in C:\Users\VssAdministrator\AppData\Local\Temp\tmp794B\Dotnet.Script\Program.cs:line 209
   at Dotnet.Script.Program.Main(String[] args) in C:\Users\VssAdministrator\AppData\Local\Temp\tmp794B\Dotnet.Script\Program.cs:line 27
```

--- a/src/Dotnet.Script.Core/ScriptPublisher.cs
+++ b/src/Dotnet.Script.Core/ScriptPublisher.cs
@@ -85,7 +85,7 @@ namespace Dotnet.Script.Core
 
             var commandRunner = new CommandRunner(logFactory);
             // todo: may want to add ability to return dotnet.exe errors
-            var exitcode = commandRunner.Execute("dotnet", $"publish \"{tempProjectPath}\" -c Release -r {runtimeIdentifier} -o {context.WorkingDirectory}");
+            var exitcode = commandRunner.Execute("dotnet", $"publish \"{tempProjectPath}\" -c Release -r {runtimeIdentifier} -o \"{context.WorkingDirectory}\"");
             if (exitcode != 0)
             {
                 throw new Exception($"dotnet publish failed with result '{exitcode}'");

--- a/src/Dotnet.Script.DependencyModel/Context/DotnetRestorer.cs
+++ b/src/Dotnet.Script.DependencyModel/Context/DotnetRestorer.cs
@@ -46,7 +46,7 @@ namespace Dotnet.Script.DependencyModel.Context
             {
                 return string.IsNullOrWhiteSpace(projectFileInfo.NuGetConfigFile)
                     ? string.Empty
-                    : $"--configfile {projectFileInfo.NuGetConfigFile}";
+                    : $"--configfile \"{projectFileInfo.NuGetConfigFile}\"";
 
             }
         }

--- a/src/Dotnet.Script.Tests/ScriptExecutionTests.cs
+++ b/src/Dotnet.Script.Tests/ScriptExecutionTests.cs
@@ -391,7 +391,7 @@ namespace Dotnet.Script.Tests
 
             using (var packageLibraryFolder = new DisposableFolder())
             {
-                CreateTestPackage(packageLibraryFolder);
+                CreateTestPackage(packageLibraryFolder.Path);
 
                 string pathToScriptFile = CreateTestScript(packageLibraryFolder.Path);
 
@@ -407,7 +407,7 @@ namespace Dotnet.Script.Tests
 
             using (var packageLibraryFolder = new DisposableFolder())
             {
-                CreateTestPackage(packageLibraryFolder);
+                CreateTestPackage(packageLibraryFolder.Path);
 
                 var scriptFolder = Path.Combine(packageLibraryFolder.Path, "ScriptFolder");
                 Directory.CreateDirectory(scriptFolder);
@@ -417,6 +417,26 @@ namespace Dotnet.Script.Tests
                 Assert.Contains("Success", result.output);
             }
         }
+
+        [Fact]
+        public void ShouldHandleLocalNuGetFileWhenPathContainsSpace()
+        {
+            TestPathUtils.RemovePackageFromGlobalNugetCache("NuGetConfigTestLibrary");
+
+            using (var packageLibraryFolder = new DisposableFolder())
+            {
+                var packageLibraryFolderPath = Path.Combine(packageLibraryFolder.Path, "library folder");
+                Directory.CreateDirectory(packageLibraryFolderPath);
+
+                CreateTestPackage(packageLibraryFolderPath);
+
+                string pathToScriptFile = CreateTestScript(packageLibraryFolderPath);
+
+                var result = ScriptTestRunner.Default.Execute($"\"{pathToScriptFile}\"");
+                Assert.Contains("Success", result.output);
+            }
+        }
+
 
         private static string CreateTestScript(string scriptFolder)
         {
@@ -429,11 +449,16 @@ WriteLine(""Success"");
             return pathToScriptFile;
         }
 
-        private static void CreateTestPackage(DisposableFolder packageLibraryFolder)
+        private static void CreateTestPackage(string packageLibraryFolder)
         {
-            ProcessHelper.RunAndCaptureOutput("dotnet", "new classlib -n NuGetConfigTestLibrary", packageLibraryFolder.Path);
-            var projectFolder = Path.Combine(packageLibraryFolder.Path, "NuGetConfigTestLibrary");
-            ProcessHelper.RunAndCaptureOutput("dotnet", $"pack -o {Path.Combine(packageLibraryFolder.Path, "packagePath")}", projectFolder);
+            ProcessHelper.RunAndCaptureOutput("dotnet", "new classlib -n NuGetConfigTestLibrary", packageLibraryFolder);
+            var projectFolder = Path.Combine(packageLibraryFolder, "NuGetConfigTestLibrary");
+            ProcessHelper.RunAndCaptureOutput("dotnet", $"pack -o \"{Path.Combine(packageLibraryFolder, "packagePath")}\"", projectFolder);
+            CreateNuGetConfig(packageLibraryFolder);
+        }
+
+        private static void CreateNuGetConfig(string packageLibraryFolder)
+        {
             string nugetConfig = @"
 <configuration>
     <packageSources>
@@ -442,7 +467,7 @@ WriteLine(""Success"");
     </packageSources>
 ></configuration>
                 ";
-            File.WriteAllText(Path.Combine(packageLibraryFolder.Path, "NuGet.Config"), nugetConfig);
+            File.WriteAllText(Path.Combine(packageLibraryFolder, "NuGet.Config"), nugetConfig);
         }
     }
 }

--- a/src/Dotnet.Script.Tests/ScriptPublisherTests.cs
+++ b/src/Dotnet.Script.Tests/ScriptPublisherTests.cs
@@ -160,18 +160,20 @@ namespace Dotnet.Script.Tests
         public void SimplePublishDllToOtherFolderTest()
         {
             using (var workspaceFolder = new DisposableFolder())
-            using (var publishFolder = new DisposableFolder())
             {
-                var code = @"WriteLine(""hello world"");";
-                var mainPath = Path.Combine(workspaceFolder.Path, "main.csx");
-                File.WriteAllText(mainPath, code);
-                var publishResult = ScriptTestRunner.Default.Execute($"publish {mainPath} --dll -o {publishFolder.Path}", workspaceFolder.Path);
-                Assert.Equal(0, publishResult.exitCode);
+                using (var publishFolder = new DisposableFolder())
+                {
+                    var code = @"WriteLine(""hello world"");";
+                    var mainPath = Path.Combine(workspaceFolder.Path, "main.csx");
+                    File.WriteAllText(mainPath, code);
+                    var publishResult = ScriptTestRunner.Default.Execute($"publish {mainPath} --dll -o {publishFolder.Path}", workspaceFolder.Path);
+                    Assert.Equal(0, publishResult.exitCode);
 
-                var dllPath = Path.Combine(publishFolder.Path, "main.dll");
-                var dllRunResult = ScriptTestRunner.Default.Execute($"exec {dllPath}", publishFolder.Path);
+                    var dllPath = Path.Combine(publishFolder.Path, "main.dll");
+                    var dllRunResult = ScriptTestRunner.Default.Execute($"exec {dllPath}", publishFolder.Path);
 
-                Assert.Equal(0, dllRunResult.exitCode);
+                    Assert.Equal(0, dllRunResult.exitCode);
+                }
             }
         }
 
@@ -219,7 +221,7 @@ namespace Dotnet.Script.Tests
         {
             using (var workspaceFolder = new DisposableFolder())
             {
-                ProcessHelper.RunAndCaptureOutput($"dotnet",$" new classlib -n MyCustomLibrary -o {workspaceFolder.Path}");
+                ProcessHelper.RunAndCaptureOutput($"dotnet", $" new classlib -n MyCustomLibrary -o {workspaceFolder.Path}");
                 ProcessHelper.RunAndCaptureOutput($"dotnet", $" build -o {workspaceFolder.Path}", workspaceFolder.Path);
                 var code = $@"#r ""MyCustomLibrary.dll"" {Environment.NewLine} WriteLine(""hello world"");";
                 var mainPath = Path.Combine(workspaceFolder.Path, "main.csx");
@@ -230,6 +232,27 @@ namespace Dotnet.Script.Tests
                 Assert.Equal(0, publishResult.exitCode);
             }
         }
+
+        [Fact]
+        public void ShouldHandleSpaceInPublishFolder()
+        {
+            using (var workspaceFolder = new DisposableFolder())
+            {
+                var code = @"WriteLine(""hello world"");";
+                var mainPath = Path.Combine(workspaceFolder.Path, "main.csx");
+                File.WriteAllText(mainPath, code);
+
+                var publishResult = ScriptTestRunner.Default.Execute(@"publish main.csx -o ""publish folder""", workspaceFolder.Path);
+
+                Assert.Equal(0, publishResult.exitCode);
+
+                var exePath = Path.Combine(workspaceFolder.Path, "publish folder", "script");
+                var executableRunResult = _commandRunner.Execute(exePath);
+
+                Assert.Equal(0, executableRunResult);
+            }
+        }
+
 
         private LogFactory GetLogFactory()
         {

